code/modules/projectiles/gun_attachables.dm

```diff
@@ -228,6 +228,8 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
  */
 /obj/item/attachable/proc/detach_from_master_gun(mob/user)
 	SHOULD_CALL_PARENT(TRUE)
+	if(!user)
+		return
 	if(flags_attach_features & ATTACH_ACTIVATION)
```

Comment:
Why?

Reply:
the loadout vendor sometimes have to replace attachments from guns. and obviously, there are no user when it does that

Reply:
but this just makes it not be removed?

Reply:
This straight delete it, which is what i want